src/core_editor/editor.rs

```diff
@@ -16,7 +26,8 @@ pub struct Editor {
     system_clipboard: Box<dyn Clipboard>,
     edit_stack: EditStack<LineBuffer>,
     last_undo_behavior: UndoBehavior,
-    selection_anchor: Option<usize>,
+    selection: Option<Selection>,
+    edit_mode: PromptEditMode,
```

Comment:
@Bahex Should we consider moving `edit_mode` from https://github.com/nushell/reedline/blob/adeb8ea52ff78917bb5f48d8904adf6cbd29bcf9/src/engine.rs#L128-L129 here?
It doesn't make much sense to me that `edit_mode` is not native to the `editor`.